### PR TITLE
Bug Fix: Not showing New Threads subscription when only subscription for community/chain

### LINKED ISSUE #617: Handle missing subscriptions for chains/communities
- Why are there no options to select here? Should be "New threads"

image

- What is "Notifications on Comments"?

image

### LINKED ISSUE #618: Handle indeterminate state for "new threads (all communities)"
The state displayed for "New threads (all communities)" here is inaccurate, since I am only subscribed to new threads in some communities

image

### PR DESCRIPTION
**Pull request by zakhap**
## Description

Closes #617, if the user only had one subscription (the community wide New Threads one), nothing would show up. 
I also made the menu link to the community's `/:scope/notificationSettings` rather than default to a chain.
Closes #618, by adding an indeterminate state to the notification subscription row.

## Motivation and Context

UI 🐛 

## How has this been tested?

Join new community and check notificationSettings specific for new community.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] no

### CODE REVIEW

File: client/scripts/views/pages/subscriptions/notificationSettings.ts [hunk 1]
```diff
@@ -99,15 +99,21 @@ const BatchedSubscriptionRow: m.Component<IBatchedSubscriptionRowAttrs, IBatched
     vnode.state.subscriptions = vnode.attrs.subscriptions;
   },
   view: (vnode) => {
+    if (vnode.state.subscriptions[0] !== vnode.attrs.subscriptions[0]) {
+      vnode.state.subscriptions = vnode.attrs.subscriptions;
+    }
```
Comment (raykyri) [inline client/scripts/views/pages/subscriptions/notificationSettings.ts:104]: What is going on here? This seems like something is off: either we shouldn't be copying attrs to state, or whatever is stored in subscriptions[0] should be handled more elegantly

File: client/scripts/views/pages/subscriptions/notificationSettings.ts [hunk 2]
```diff
@@ -191,19 +197,23 @@ const CommunitySpecificNotifications: m.Component<ICommunitySpecificNotification
         && s.category !== NotificationCategories.ChainEvent
         && !s.OffchainComment
     );
+    const newThreads = subscriptions.find(
+      (s) => (s.category === NotificationCategories.NewThread && s.objectId === community.id)
+    );
+    console.log('new threads', newThreads);
```
Comment (raykyri) [inline client/scripts/views/pages/subscriptions/notificationSettings.ts:203]: Remove

File: client/scripts/views/pages/subscriptions/notificationSettings.ts (hunk 1 above)
Comment (zakhap) [reply client/scripts/views/pages/subscriptions/notificationSettings.ts:104]: handled this fix! you were right, there was no need to copy attrs into state here.

**Comment by zakhap** [comment]
I merged this over into PR #669

**Pull request closed without merging**